lib/bootstrap_form/form_builder.rb

```diff
@@ -245,6 +246,32 @@ def fields_for_with_bootstrap(record_name, record_object = nil, fields_options =
 
     bootstrap_method_alias :fields_for
 
+    # Add bootstrap formatted submit button. If you need to change its type or
+    # add another css class, you need to override all css classes like so:
+    #
+    #   <%= form.submit class: "btn btn-info custom-class" %>
+    #
+    # You may add additional content that directly follows the button. Here's
+    # an example of a cancel link:
+    #
+    #   <%= form.submit do %>
+    #     <%= link_to "Cancel", "/", class: "btn btn-link" %>
+    #   <% end %>
+    #
+    def submit(value = nil, options = {}, &block)
+      out = super(value, options.reverse_merge(class: "btn"))
+      out += capture(&block) if block_given?
+
+      form_group do
```

Comment:
Sorry my earlier comment wasn't complete. If we're adding a `div.form-group` to the markup for a submit button, this may or may not change the layout of some pages. I ran into a lot of cases where the addition of even one `div` would change the way the browser actually renders Bootstrap 4 markup. I resorted to trying just about everything in Bootply, comparing the Bootstrap 3/`bootstrap_form` v2.7 rendering to the Bootstrap 4/`bootstrap_form` v4.0 rendering. Tedious, but might avoid some surprises.

Reply:
There's a /demo here where you can try out all types of scenarios. Easier than copy-pasting code into Bootply.